articles/group-operator/group.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/group-operator-in-mongodb/
+title: Group operator in MongoDB
+description:  Group operator is a very important operator in MongoDB language and it is a part of aggregation in MongoDB so this article will cover $group operator, its applications along with examples.
+author: harit-joshi
+date: 21:52:42 2021 +0530
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/group-operator/hero.jpg
+    alt: Group operator in MongoDB Image
+---
+Group operator is a very important operator in MongoDB language and it is a part of aggregation in MongoDB.
+
+In this article we are going to learn:
+
+1. What is the $group operator?
+
+2. Examples for a better understanding of the operator.
+
+3. Applications of $group operator.
+
+## Aggregation
+
+Aggregation is an operation that processes the data to give a computed result. It basically involves stages or pipelines through which data is processed to yield a combined result. So pipelines are stages through which data is processed, more technically transformed into by the provided criteria.
+ 
+Each pipeline is independent and receives the data of the previous stage and always remember that always the first pipeline will have direct contact with the data itself, not its transformed version, and can take advantage of indexes.
+
+## What is $group operator?
+
+So $group operator as the name suggests groups the similar data by some specified expression and combines it into a single result or document. Just imagine like this suppose there are 15 people and they all have a similar hobby and if we say count all those people who have a similar hobby so let’s say 3 people have swimming as a similar hobby, 5 people have cooking as a similar hobby and so on.
```

Comment:
```suggestion
As the name suggests, the `$group` operator groups similar data by some specified expression and combines it into a single result or document. 

Suppose there are 15 people and they all have a similar hobby and if we count all those people who have a similar hobby so let’s say 3 people have swimming as a similar hobby, 5 people have cooking as a similar hobby and so on.
```

Reply:
`Suppose there are 15 people and they all have a similar hobby and if we count all those people who have a similar hobby so let’s say 3 people have swimming as a similar hobby, 5 people have cooking as a similar hobby and so on.`

It's not clear what you are conveying in this line. It's too long. Break it down properly so that it will be easier to read.